app/controllers/api/plans_base_controller.rb

```diff
@@ -28,8 +28,9 @@ def authorize_action
     authorize! :create, :plans
   end
 
-  def resource(id = params[:id])
-    collection.readonly(false).find_by_id(id)
+  def resource(id = params[:id].presence)
+    return unless id
+    collection.readonly(false).find(id)
   end
```

Comment:
You wanted to do probably this:

```suggestion
  def resource(id = params[:id])
    return unless id.present?
    collection.readonly(false).find(id)
  end
```

Because in your version if you call `resource("")` it will still execute `collection.readonly(false).find("")`

Reply:
@guicassolato